joomla/joomla-cms
# Problems with tinyMCE width and height, joomla 3.7

I think there are some confusions about the computed values of the width and height values for the tinyMCE editor. Personally I categorize this as a bug. 

In theory there could be a lot of default settings but this is more simple.

We get the values from user editable system wide parameters (with some funny results like height=750, the tinyMCE default )):
- tinyMCE editor default values
- tinyMCE editor settings
- this overrides the parameters from the form in all components and views.

Article edit
![article_is](https://cloud.githubusercontent.com/assets/956537/24589129/7f223242-17d5-11e7-8a67-1072461e232c.jpg)

A second field in article edit as in PR 14520.
![article_is2](https://cloud.githubusercontent.com/assets/956537/24589143/bca5d704-17d5-11e7-9eeb-bece201a535d.jpg)

Article frontend editing.
![article frontend](https://cloud.githubusercontent.com/assets/956537/24589167/3037855a-17d6-11e7-9f5d-75ec6a1fbcf9.jpg)

The Joomla editor form field will never get any settings from external sources as default.

My propasal is to set the width hardcoded to 100% (or auto) and the height from:
- Joomla editor form field default settings
- the field's settings in the form field XML
- tinyMCE default setting in the code (if no value available). I'm not quite sure about this.
- override in a system plug in - onBeforeRender() method

Expected result
![article_should](https://cloud.githubusercontent.com/assets/956537/24589179/791c8de2-17d6-11e7-8f9a-9a2ab1ccf819.jpg)

As tinyMCE automatically gets the height values from the hidden textarea field, we don't have to explicit set any values in the tinyMCE options. As well, we don't have to set any editor instance values as in #14520.

If you want a width per field, this has to be added to each editor instance as in PR #14520. As far as I could see there is no automatic loading as with the height.

In this PR I have removed the parameters in the tinyMCE editor, removed the tinyMCE options, added defaults in tinyMCE code and deprecated the language strings. I did not change the legacy part. Width is now overridable with a custom template/layout and height by using a custom system plugin - method onBeforerender().

Test instructions:
1. Add an extra editor field as in #14520 with height. The original article text field has no expicit value, default 500px height from the form field.

In the article form XML administrator/components/com_content/models/forms/article.xml add one more editor element (somewhere after line 
`<fieldset name="basic" label="COM_CONTENT_ATTRIBS_FIELDSET_LABEL">`)

`<field name="text2" type="editor" hide="menu,module,contact"
 label="Text2" description="" filter="JComponentHelper::filterText" buttons="true" height="250"/>`

2. Check the result without adding the patch. Play with the tinyMCE editor settings a bit. Go to the tab(s) advanced in tinyMCE and modify Width and Height to what a user could enter. e.g. width 50%, height=200.
Have a look to the new second field in article edit, backend categories and frontend article edit/creation as well. 

3. Apply this patch.
There are no settings so just reload the article edit page and .
Check and compare the results.

@Fedik  Is it easy to add the width if !100% as in PR 14520? The width could then be set in the form field. Needed? 

Related
#14917
#14835
